src/ArborX_DistributedSearchTree.hpp

```diff
@@ -120,11 +120,14 @@ DistributedSearchTree<MemorySpace, Enable>::DistributedSearchTree(
   // Create new context for the library to isolate library's communication from
   // user's
   _comm_ptr.reset(
+      // duplicate the communicator and store it in a std::shared_ptr so that
+      // all copies of the distributed tree point to the same object
       [comm]() {
         auto p = std::make_unique<MPI_Comm>();
```

Comment:
I actually wanted to see a comment on why `unique_ptr` is necessary here.

Reply:
Manages lifetime of the pointer until ownership is transferred to the `std::shared_ptr` (would call `delete` if anything goes wrong and an exception is thrown before the `shared_ptr` is constructed)

Reply:
From conversation with @dalg24, the point is that if we create a pointer and then something happens inside `MPI_Comm_dup()` in that lambda, the pointer is never deleted. Wrapping it in the unique pointer takes care of that.

Reply:
I don't feel strongly enough about this whole issue to argue against anything the two of you have agreed upon. If we are that concerned about errors in MPI calls, checking the return values feels appropriate, though.

Reply:
>  If we are that concerned about errors in MPI calls, checking the return values feels appropriate, though.

This is a larger issue.  The user may have specified an error handler that throws.